packages/node/src/channels.ts

```diff
@@ -0,0 +1,387 @@
+import { legacy } from "@counterfactual/cf.js";
+import {
+  Address,
+  AppInstanceInfo,
+  AssetType,
+  Node
+} from "@counterfactual/common-types";
+import { ethers } from "ethers";
+
+import { IStoreService } from "./service-interfaces";
+import { zeroBalance } from "./utils";
+
+import Nonce = legacy.utils.Nonce;
+import FreeBalance = legacy.utils.FreeBalance;
+
+/**
+ * The schema of a channel is below.
+ * The following Channels class encapsulates the state and persistence of all
+ * the channels in a Node.
+ */
+
+/**
+ * The fully expanded schema for a channel:
+ * multisigAddress: {
```

Comment:
I assume this isn't a literal string - does this mean "keyed by multisigAddress"? what is a schema - is it a firebase concept?

Reply:
Yes, keyed by multisig address.

What is the object representation encapsulating the concept of a state channel?

Reply:
> What is the object representation encapsulating the concept of a state channel?

Are you answering me in Jeopardy! syntax?

Reply:
hmm, can we write "firebase schema" (maybe link to a reference) early on in the comments before abbreviating it to "schema"

Reply:
What makes this inherently tied to firebase?

Reply:
Any key/value store should be able to store an object with this representation, no?

Reply:
It wasn't clear to me whether "schema" is meant to refer to a firebase-specific concept or a more general concept